[pointclouds] Part 1: General infrastructure of the new layer type and 2d/3d renderer

PeterPetrik opened the pull request:
This PR adds the general ground for work towards support of point cloud rendering in QGIS. 

- Adds new QgsPointCloudLayer
- Adds new QgsPointCloudDataProvider and the very rough implementation of EPT format 
- Adds a very basic 2D point cloud renderer, without GUI settings
- Adds a very basic 3D point cloud renderer, without GUI settings
- Adds a simple 3D sandbox executable, if you have qgis project with 3d layers, you can load it directly via sandbox 3D
- Adds a basic integration of QgsPointCloudLayer into browser and basic data manager dialog

Lacks almost everything else. Please consider this really as a somewhat functional stubs of classes that we will work on in the following weeks. Test are missing. The implementation of various parts is really just placeholder (e.g EPT reader, rendering...)

<img width="827" alt="Screenshot 2020-10-22 at 16 19 02" src="https://user-images.githubusercontent.com/804608/97040424-31249900-156e-11eb-8874-555a0acc41ed.png">

See https://github.com/qgis/QGIS-Enhancement-Proposals/issues/194 for overall technical details
This work is possible because of generous support from community, see https://www.lutraconsulting.co.uk/crowdfunding/pointcloud-qgis/

PeterPetrik committed:
initial files stubs

PeterPetrik committed:
hook point cloud layer type to the rest of the code

PeterPetrik committed:
bunch of other file skeletons

PeterPetrik committed:
add simple browser integration

PeterPetrik committed:
port the basic renderer from Martin's prototype

PeterPetrik committed:
read CSR and bounding box

wonder-sk committed:
Fixed few bugs + minor rendering improvements

- using a color ramp to render points
- only rendering nodes that intersect the map extent

wonder-sk committed:
Take point record size into account when decoding

PeterPetrik committed:
some stubs for 3d classes

PeterPetrik committed:
fix after Nyall review

PeterPetrik committed:
port academic code from Martin's prototype. ugly me

wonder-sk committed:
Random fixes and hacks

- corrected rotation
- hack to make the cloud closer to the terrain
- programmatic sizing of points
- limit to a single root chunk

PeterPetrik committed:
add some basic skelet for point cloud data providers

wonder-sk committed:
Fix qgis_core python and project loading with point cloud layer

wonder-sk committed:
Added a qgis 3d test executable to quickly open a 3D scene

The 3d sandbox needs a qgis project file and will open 3D map canvas

wonder-sk committed:
Fix entity's bbox and always show the bboxes for point clouds (for now)

wonder-sk committed:
fix loading of classes

PeterPetrik committed:
add ept provider

PeterPetrik committed:
able to build without zstd and lazperf

PeterPetrik committed:
restore accidentally deleted files

PeterPetrik committed:
fix loading

nyalldawson commented on python/3d/auto_generated/qgspointcloudlayer3drenderer.sip.in:
```diff
@@ -0,0 +1,91 @@
+
+
+
+class QgsPointCloudLayer3DRendererMetadata : Qgs3DRendererAbstractMetadata
```
This class can be ommited from sip - it's an internal detail only

nyalldawson commented on python/core/auto_generated/pointcloud/qgspointcloudlayer.sip.in:
```diff
@@ -0,0 +1,102 @@
+Constructor for LayerOptions with optional ``transformContext``.
+
+.. note::
+
```
Remove this note

nyalldawson commented on python/core/auto_generated/pointcloud/qgspointcloudrenderer.sip.in:
```diff
@@ -0,0 +1,88 @@
+
+
+
+class QgsPointCloudRendererConfig
```
Internal detail, skip from sip

nyalldawson commented on src/3d/qgspointcloudlayer3drenderer.h:
```diff
@@ -0,0 +1,86 @@
+    //! Returns point cloud layer associated with the renderer
+    QgsPointCloudLayer *layer() const;
+
+    QString type() const override { return "pointcloud"; }
```
```suggestion
    QString type() const override { return QStringLiteral( "pointcloud" ); }
```

nyalldawson commented on src/core/pointcloud/qgspointclouddataprovider.h:
```diff
@@ -0,0 +1,62 @@
+
+    bool isValid() const override;
+
+    virtual QString name() const override = 0;
```
These should all be removed - I don't see the point in overriding them with another pure virtual method?

nyalldawson committed:
Fix some warnings

nyalldawson committed:
Implement QgsPointCloudLayer::setTransformContext

nyalldawson committed:
Fix warning

nyalldawson committed:
Fix more warnings

nyalldawson committed:
Fix typo

nyalldawson committed:
More warning fixes

nyalldawson committed:
Remove QgsPointCloudLayer3DRendererMetadata from sip

nyalldawson committed:
Update note

nyalldawson committed:
Minor api cleanups

nyalldawson committed:
Mark QgsPointCloudRendererConfig as private for now

nyalldawson committed:
QStringLiteral

nyalldawson committed:
QgsPointCloudDataBounds is private api for now

nyalldawson committed:
Remove write/readXml from QgsPointCloudRenderer -- these don't belong in a QgsMapLayerRenderer subclass

nyalldawson committed:
Rename QgsPointCloudRenderer to QgsPointCloudLayerRenderer

This follow the pattern of QgsVectorLayerRenderer, QgsRasterLayerRenderer,
etc and disambiguates the layer renderer from the 2d renderer
itself.

nyalldawson committed:
Fix build

nyalldawson committed:
QgsMapLayerRenderer subclasses aren't exposed to python

nyalldawson committed:
QgsMapLayerRenderers must be created on demand, not stored

nyalldawson committed:
Remove unnecessary copy constructor from QgsPointCloudDataBounds -- it is identical to implicit one (fixes cppcheck warning)

nyalldawson committed:
Set data provider key for QgsPointCloudLayer, so that QgsPointCloudLayer::providerType()
returns the correct value

Fixes a crash on exit caused because the provider metadata for the layer
cannot be retrieved unless providerType is correct

nyalldawson committed:
Minimal encode/decode uri implementations for ept data provider

PeterPetrik committed:
Merge pull request #23 from nyalldawson/pointclouds

Some fixes

nyalldawson committed:
Make setting point cloud layer data source follow expected behavior
for map layers

Will make it possible to repair point cloud layer sources

nyalldawson committed:
Fix cloning point cloud layers

nyalldawson committed:
Point cloud layer data sources can be changed via menu

nyalldawson committed:
Fix build

PeterPetrik committed:
fix some checks and building

PeterPetrik committed:
fix builds

PeterPetrik committed:
fix builds

PeterPetrik committed:
fix build

PeterPetrik committed:
fix builds

PeterPetrik committed:
fix travis

PeterPetrik committed:
fix travis

PeterPetrik committed:
fix travis

PeterPetrik committed:
fix travis and 5.14 builds

wonder-sk committed:
Fix build of sandbox + fix loading/saving project

wonder-sk committed:
Chunk node refactor: remove tileX(), tileY(), tileZ()

Let's use tileId() everywhere for chunk coordinates

wonder-sk committed:
Octree support in chunk nodes, start using chunking for point clouds

nyalldawson committed:
Fix missing bad layer indicator for point cloud/mesh layers

nyalldawson committed:
Basic layer properties page for point cloud layers

nyalldawson committed:
Show information in point cloud layer properties

nyalldawson committed:
Fix display of point cloud layer properties/preview in browser

nyalldawson committed:
Enabled runtime profiling of ept provider creation

nyalldawson committed:
Add hacky way to configure 2d renderer via layer custom properties for now

PeterPetrik committed:
fix travis and docs

nyalldawson committed:
Move point cloud placement in data source manager

PeterPetrik commented:
@wonder-sk some 3d tests fail, maybe can you look at it so we can be ready  for merging?

wonder-sk committed:
Refactor QgsChunkNode + QgsChunkedEntity for more flexibility

- QgsChunkNode can have variable number of children
- QgsChunkLoaderFactory implementations get responsibility to return
  root node and child nodes of any node in the hierarchy
- new QgsQuadtreeChunkLoaderFactory class to implement quadtree strategy
- point cloud chunk loader updated to only return children that exist in data

PeterPetrik committed:
fix doc check

wonder-sk committed:
Fix inverted tiling

wonder-sk commented:
Tests are finally green again :tada: 

From my point of view, I am happy with all the code changes from @PeterPetrik and @nyalldawson - if others do not have objections it would be good to merge this very soon before it starts to diverge from master branch - it is already over 5K lines of new code.

PeterPetrik commented:
I am happy to merge too!

nyalldawson commented:
Merge away!!

nirvn commented:
@wonder-sk , @nyalldawson , @PeterPetrik , me can't compile anymore, sip throws the following errors:
```

CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `QByteArray::QByteArray()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:176: multiple definition of `sipQgs3DMapSettings::sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1152: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `QByteArray::QByteArray()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:176: multiple definition of `sipQgs3DMapSettings::sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1152: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipQgs3DMapSettings(Qgs3DMapSettings const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:181: multiple definition of `sipQgs3DMapSettings::sipQgs3DMapSettings(Qgs3DMapSettings const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1157: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipQgs3DMapSettings(Qgs3DMapSettings const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:181: multiple definition of `sipQgs3DMapSettings::sipQgs3DMapSettings(Qgs3DMapSettings const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1157: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::~sipQgs3DMapSettings()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:186: multiple definition of `sipQgs3DMapSettings::~sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1162: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `Qgs3DMapSettings::Qgs3DMapSettings()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:133: multiple definition of `non-virtual thunk to sipQgs3DMapSettings::~sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1109: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::~sipQgs3DMapSettings()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:186: multiple definition of `sipQgs3DMapSettings::~sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1162: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::~sipQgs3DMapSettings()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:186: multiple definition of `sipQgs3DMapSettings::~sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1162: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `Qgs3DMapSettings::Qgs3DMapSettings()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:133: multiple definition of `non-virtual thunk to sipQgs3DMapSettings::~sipQgs3DMapSettings()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1109: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::metaObject() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:192: multiple definition of `sipQgs3DMapSettings::metaObject() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1168: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::qt_metacall(QMetaObject::Call, int, void**)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:200: multiple definition of `sipQgs3DMapSettings::qt_metacall(QMetaObject::Call, int, void**)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1176: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::qt_metacast(char const*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:214: multiple definition of `sipQgs3DMapSettings::qt_metacast(char const*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1190: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::event(QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:221: multiple definition of `sipQgs3DMapSettings::event(QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1197: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::eventFilter(QObject*, QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:236: multiple definition of `sipQgs3DMapSettings::eventFilter(QObject*, QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1212: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::timerEvent(QTimerEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:251: multiple definition of `sipQgs3DMapSettings::timerEvent(QTimerEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1227: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::childEvent(QChildEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:269: multiple definition of `sipQgs3DMapSettings::childEvent(QChildEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1245: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::customEvent(QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:287: multiple definition of `sipQgs3DMapSettings::customEvent(QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1263: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::connectNotify(QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:305: multiple definition of `sipQgs3DMapSettings::connectNotify(QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1281: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::disconnectNotify(QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:323: multiple definition of `sipQgs3DMapSettings::disconnectNotify(QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1299: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtect_sender() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:341: multiple definition of `sipQgs3DMapSettings::sipProtect_sender() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1317: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtect_receivers(char const*) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:346: multiple definition of `sipQgs3DMapSettings::sipProtect_receivers(char const*) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1322: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtectVirt_timerEvent(bool, QTimerEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:351: multiple definition of `sipQgs3DMapSettings::sipProtectVirt_timerEvent(bool, QTimerEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1327: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtectVirt_childEvent(bool, QChildEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:356: multiple definition of `sipQgs3DMapSettings::sipProtectVirt_childEvent(bool, QChildEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1332: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtectVirt_customEvent(bool, QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:361: multiple definition of `sipQgs3DMapSettings::sipProtectVirt_customEvent(bool, QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1337: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtectVirt_connectNotify(bool, QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:366: multiple definition of `sipQgs3DMapSettings::sipProtectVirt_connectNotify(bool, QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1342: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtectVirt_disconnectNotify(bool, QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:371: multiple definition of `sipQgs3DMapSettings::sipProtectVirt_disconnectNotify(bool, QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1347: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtect_senderSignalIndex() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:376: multiple definition of `sipQgs3DMapSettings::sipProtect_senderSignalIndex() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1352: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DMapSettings::sipProtect_isSignalConnected(QMetaMethod const&) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:381: multiple definition of `sipQgs3DMapSettings::sipProtect_isSignalConnected(QMetaMethod const&) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:1357: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:3671: multiple definition of `sipTypeDef__3d_Qgs3DMapSettings'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart7.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart7.cpp:4647: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:3949: multiple definition of `sipTypeDef__3d_Qgs3D'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:170: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipQgs3DAlgorithms(QObject*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4151: multiple definition of `sipQgs3DAlgorithms::sipQgs3DAlgorithms(QObject*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:372: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipQgs3DAlgorithms(QObject*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4151: multiple definition of `sipQgs3DAlgorithms::sipQgs3DAlgorithms(QObject*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:372: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4156: multiple definition of `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:377: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4156: multiple definition of `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:377: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4156: multiple definition of `sipQgs3DAlgorithms::~sipQgs3DAlgorithms()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:377: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::metaObject() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4162: multiple definition of `sipQgs3DAlgorithms::metaObject() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:383: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::qt_metacall(QMetaObject::Call, int, void**)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4170: multiple definition of `sipQgs3DAlgorithms::qt_metacall(QMetaObject::Call, int, void**)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:391: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::qt_metacast(char const*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4184: multiple definition of `sipQgs3DAlgorithms::qt_metacast(char const*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:405: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::disconnectNotify(QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4191: multiple definition of `sipQgs3DAlgorithms::disconnectNotify(QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:412: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::connectNotify(QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4209: multiple definition of `sipQgs3DAlgorithms::connectNotify(QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:430: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::customEvent(QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4227: multiple definition of `sipQgs3DAlgorithms::customEvent(QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:448: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::childEvent(QChildEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4245: multiple definition of `sipQgs3DAlgorithms::childEvent(QChildEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:466: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::timerEvent(QTimerEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4263: multiple definition of `sipQgs3DAlgorithms::timerEvent(QTimerEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:484: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::eventFilter(QObject*, QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4281: multiple definition of `sipQgs3DAlgorithms::eventFilter(QObject*, QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:502: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::event(QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4296: multiple definition of `sipQgs3DAlgorithms::event(QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:517: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::icon() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4311: multiple definition of `sipQgs3DAlgorithms::icon() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:532: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::svgIconPath() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4326: multiple definition of `sipQgs3DAlgorithms::svgIconPath() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:547: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::flags() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4341: multiple definition of `sipQgs3DAlgorithms::flags() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:562: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::id() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4356: multiple definition of `sipQgs3DAlgorithms::id() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:577: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::helpId() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4371: multiple definition of `sipQgs3DAlgorithms::helpId() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:592: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::name() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4386: multiple definition of `sipQgs3DAlgorithms::name() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:607: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::longName() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4401: multiple definition of `sipQgs3DAlgorithms::longName() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:622: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::versionInfo() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4416: multiple definition of `sipQgs3DAlgorithms::versionInfo() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:637: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::canBeActivated() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4431: multiple definition of `sipQgs3DAlgorithms::canBeActivated() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:652: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::warningMessage() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4446: multiple definition of `sipQgs3DAlgorithms::warningMessage() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:667: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::isActive() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4461: multiple definition of `sipQgs3DAlgorithms::isActive() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:682: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::supportedOutputRasterLayerExtensions() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4476: multiple definition of `sipQgs3DAlgorithms::supportedOutputRasterLayerExtensions() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:697: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::supportedOutputVectorLayerExtensions() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4491: multiple definition of `sipQgs3DAlgorithms::supportedOutputVectorLayerExtensions() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:712: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::supportedOutputTableExtensions() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4506: multiple definition of `sipQgs3DAlgorithms::supportedOutputTableExtensions() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:727: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::isSupportedOutputValue(QVariant const&, QgsProcessingDestinationParameter const*, QgsProcessingContext&, QString&) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4521: multiple definition of `sipQgs3DAlgorithms::isSupportedOutputValue(QVariant const&, QgsProcessingDestinationParameter const*, QgsProcessingContext&, QString&) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:742: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::defaultVectorFileExtension(bool) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4536: multiple definition of `sipQgs3DAlgorithms::defaultVectorFileExtension(bool) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:757: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::defaultRasterFileExtension() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4551: multiple definition of `sipQgs3DAlgorithms::defaultRasterFileExtension() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:772: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::supportsNonFileBasedOutput() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4566: multiple definition of `sipQgs3DAlgorithms::supportsNonFileBasedOutput() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:787: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::load()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4581: multiple definition of `sipQgs3DAlgorithms::load()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:802: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::unload()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4596: multiple definition of `sipQgs3DAlgorithms::unload()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:817: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::loadAlgorithms()':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4614: multiple definition of `sipQgs3DAlgorithms::loadAlgorithms()'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:835: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtect_sender() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4632: multiple definition of `sipQgs3DAlgorithms::sipProtect_sender() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:853: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtect_receivers(char const*) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4637: multiple definition of `sipQgs3DAlgorithms::sipProtect_receivers(char const*) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:858: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_timerEvent(bool, QTimerEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4642: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_timerEvent(bool, QTimerEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:863: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_childEvent(bool, QChildEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4647: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_childEvent(bool, QChildEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:868: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_customEvent(bool, QEvent*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4652: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_customEvent(bool, QEvent*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:873: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_connectNotify(bool, QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4657: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_connectNotify(bool, QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:878: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_disconnectNotify(bool, QMetaMethod const&)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4662: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_disconnectNotify(bool, QMetaMethod const&)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:883: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtect_senderSignalIndex() const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4667: multiple definition of `sipQgs3DAlgorithms::sipProtect_senderSignalIndex() const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:888: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtect_isSignalConnected(QMetaMethod const&) const':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4672: multiple definition of `sipQgs3DAlgorithms::sipProtect_isSignalConnected(QMetaMethod const&) const'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:893: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtect_addAlgorithm(QgsProcessingAlgorithm*)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4677: multiple definition of `sipQgs3DAlgorithms::sipProtect_addAlgorithm(QgsProcessingAlgorithm*)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:898: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o: in function `sipQgs3DAlgorithms::sipProtectVirt_loadAlgorithms(bool)':
/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:4682: multiple definition of `sipQgs3DAlgorithms::sipProtectVirt_loadAlgorithms(bool)'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:903: first defined here
/usr/bin/ld: CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart10.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart10.cpp:5528: multiple definition of `sipTypeDef__3d_Qgs3DAlgorithms'; CMakeFiles/python_module_qgis__3d.dir/3d/sip_3dpart8.cpp.o:/home/webmaster/dev/cpp/bm-qt5/python/3d/sip_3dpart8.cpp:1749: first defined here

```

nyalldawson commented:
@nirvn wipe your build/python folder and rebuild.

nirvn commented:
Alright, here goes 55 minutes.

PeterPetrik commented:
also "have you tried to turn it off and on again"?
